components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/AbstractAPIManager.java

```diff
@@ -1908,7 +1908,17 @@ public boolean isDuplicateContextTemplate(String contextTemplate) throws APIMana
             // If the query does not contains "=" then it is an errornous scenario.
             if (query.contains("=")) {
                 String[] searchKeys = query.split("=");
-                if (!Arrays.asList(APIConstants.API_SEARCH_PREFIXES).contains(searchKeys[0].toLowerCase())) {
+
+                //prevent api-meta. getting prefixed to labelName and restrict label serach to exact match only
+                if (APIConstants.LABELNAME.equals(searchKeys[0])) {
+                    searchKeys[0] = APIConstants.LABEL_PREFIX + searchKeys[0];
+                    if (searchKeys[1].startsWith("*")) {
```

Comment:
There's a possible array index out of bonds issue here, if the searchKeys array does not have 2 or more items.

Reply:
fixed with https://github.com/wso2/carbon-apimgt/pull/5551